File: first-unique-character-in-a-string/first-unique-character-in-a-string-01.cpp
#include<iostream>
#include<vector>

#define MAX_LEN 100010

using namespace std;

class Solution {
public:
  int firstUniqChar(string s) {
    vector<int> hash(200, 0);

    for(int i = 0; i < s.size(); i++)
        hash[s[i]] += 1;

    for(int i = 0; i < s.size(); i++){
      if(hash[s[i]] == 1){
        return i;
      }
    }

    return -1;
  }
};

int main(){
    string word;
    Solution solution;

    cin >> word;

    cout << solution.firstUniqChar(word);
    return 0;
}

File: first-unique-character-in-a-string/first-unique-character-in-a-string-01.rb
# @param {String} s
# @return {Integer}
def first_uniq_char(s)
  hash = {}

  s.each_char do |c|
    hash[c] ||= 0
    hash[c] += 1
  end

  s.each_char.with_index { |c, i| return i if hash[s[i]] == 1 }

  -1
end

word = gets.chomp
puts first_uniq_char(word)
